SU2_CFD/include/gradients/computeGradientsGreenGauss.hpp

```diff
@@ -70,81 +83,148 @@ void computeGradientsGreenGauss(CSolver* solver,
-        gradient(iPoint, iVar, iDim) = 0.0;
+      for (size_t iDim = 0; iDim < nDim; ++iDim) gradient(iPoint, iVar, iDim) = 0.0;
 
     /*--- Handle averaging and division by volume in one constant. ---*/
 
-    su2double halfOnVol = 0.5 / (nodes->GetVolume(iPoint)+nodes->GetPeriodicVolume(iPoint));
+    su2double halfOnVol = 0.5 / (nodes->GetVolume(iPoint) + nodes->GetPeriodicVolume(iPoint));
 
     /*--- Add a contribution due to each neighbor. ---*/
 
-    for (size_t iNeigh = 0; iNeigh < nodes->GetnPoint(iPoint); ++iNeigh)
-    {
-      size_t iEdge = nodes->GetEdge(iPoint,iNeigh);
-      size_t jPoint = nodes->GetPoint(iPoint,iNeigh);
+    for (size_t iNeigh = 0; iNeigh < nodes->GetnPoint(iPoint); ++iNeigh) {
+      size_t iEdge = nodes->GetEdge(iPoint, iNeigh);
+      size_t jPoint = nodes->GetPoint(iPoint, iNeigh);
 
       /*--- Determine if edge points inwards or outwards of iPoint.
        *    If inwards we need to flip the area vector. ---*/
 
-      su2double dir = (iPoint < jPoint)? 1.0 : -1.0;
+      su2double dir = (iPoint < jPoint) ? 1.0 : -1.0;
       su2double weight = dir * halfOnVol;
 
       const auto area = geometry.edges->GetNormal(iEdge);
       AD::SetPreaccIn(area, nDim);
 
-      for (size_t iVar = varBegin; iVar < varEnd; ++iVar)
-      {
-        AD::SetPreaccIn(field(jPoint,iVar));
-
-        su2double flux = weight * (field(iPoint,iVar) + field(jPoint,iVar));
+      for (size_t iVar = varBegin; iVar < varEnd; ++iVar) {
+        AD::SetPreaccIn(field(jPoint, iVar));
+        su2double flux = weight * (field(iPoint, iVar) + field(jPoint, iVar));
 
-        for (size_t iDim = 0; iDim < nDim; ++iDim)
-          gradient(iPoint, iVar, iDim) += flux * area[iDim];
+        for (size_t iDim = 0; iDim < nDim; ++iDim) gradient(iPoint, iVar, iDim) += flux * area[iDim];
       }
-
     }
 
     for (size_t iVar = varBegin; iVar < varEnd; ++iVar)
-      for (size_t iDim = 0; iDim < nDim; ++iDim)
-        AD::SetPreaccOut(gradient(iPoint,iVar,iDim));
+      for (size_t iDim = 0; iDim < nDim; ++iDim) AD::SetPreaccOut(gradient(iPoint, iVar, iDim));
 
     AD::EndPreacc();
   }
   END_SU2_OMP_FOR
 
-  /*--- Add boundary fluxes. ---*/
+  /* For symmetry planes, we need to impose the conditions (Blazek eq. 8.40):
+   * 1. n.grad(phi) = 0
+   * 2. n.grad(v.t) = 0
+   * 3. t.grad(v.n) = 0
+   */
+  su2double flux[MAXNVAR] = {0.0};
+  su2double fluxReflected[MAXNVAR] = {0.0};
+
+  for (size_t iMarker = 0; iMarker < geometry.GetnMarker(); ++iMarker) {
+    if (config.GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE) {
+      for (size_t iVertex = 0; iVertex < geometry.GetnVertex(iMarker); ++iVertex) {
+
+        size_t iPoint = geometry.vertex[iMarker][iVertex]->GetNode();
+        cout << "iPoint = " << iPoint << endl;
+        auto nodes = geometry.nodes;
+        // we need to set the gradient to zero for the entire marker to prevent double-counting
+        // points that are shared by other markers
```

Comment:
Did you have issues without this? I expected we could modify multiple times since the modification is based on removing the projection of the gradient in certain direction, so once it's removed there's no issue in trying to remove again since the projection will be 0.

Reply:
It is necessary for the stand-alone GG gradient correction. It might not be necessary when the normals and fluxes are corrected.

Reply:
Consider a quadrilateral. We only modify the side edges. We still have contributions to the gradient in both directions from the top edge, so we still need to mirror the cell or do something else to cancel contributions normal to the symmetry plane.

Reply:
If you do g' = g - g.n n you ensure that g'.n = 0. The modification can be done to the entire gradient, you don’t have to go face by face.